File: myapp/lib/Models/Note.dart
class Note{
  int Id_Note;
  int Id_Mat;
  String N_Mat;
  double note;
  String Annee="";
  String Niveau="";
  String Parcours="";

  Note(this.Id_Note,this.Id_Mat, this.N_Mat, this.note, this.Annee, this.Niveau,
      this.Parcours);

  Map toJson() => {
    'Id_Note':Id_Note,
    'Id_Mat':Id_Mat,
    'N_mat':N_Mat,
    'note':note,
    'Annee':Annee,
    'Niveau':Niveau,
    'Parcours':Parcours,
  };

}

File: gestion_note/app/Models/Note.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Note extends Model
{
    use HasFactory;

    protected $table = "note";

    protected $primaryKey = 'Id_Note';
    protected $fillable = ['Id_Mat', 'N_mat', 'note', 'Annee', 'Niveau', 'Parcours'];
}
